Starwatch.Core/API/Rest/Route/AnnouncementRoute.cs

```diff
@@ -0,0 +1,387 @@
+            {
+                return new RestResponse(RestStatus.InternalError, NO_MONITOR);
+            }
+
+            if (query.ContainsKey("id"))
+            {
+                int id = 0;
+                bool ok = int.TryParse(query["id"], out id);
+
+                if (!ok)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_INTEGER);
+                }
+
+                if (id < 0)
+                {
+                    return new RestResponse(RestStatus.BadRequest, ID_MUST_BE_POSITIVE);
+                }
+
+                if (!mon.Lock())
+                {
+                    return new RestResponse(RestStatus.InternalError, RESOURCE_LOCKED);
+                }
+
+                if (mon.Announcements.Length > id)
+                {
+                    try
+                    {
+                        AnnouncementMonitor.Announcement target = mon.Announcements[id];
+
+                        mon.Announcements = mon.Announcements.Where((s, i) => i != id).ToArray();
+                        mon.Configuration.SetKey("announcements", mon.Announcements);
+                        mon.Configuration.Save();
+
+                        mon.Unlock();
+                        mon.Reload();
+
+                        return new RestResponse(RestStatus.OK, $"Deleted announcement #{id}", target);
+                    }
+                    catch (Exception ex)
+                    {
+                        mon.Unlock();
+                        return new RestResponse(RestStatus.InternalError, "Failed to save config: " + ex.Message);
+                    }
+                }
+
+                mon.Unlock();
+                return new RestResponse(RestStatus.ResourceNotFound, MISSING_RESOURCE);
+            }
+            else
+            {
+                return new RestResponse(RestStatus.BadRequest, "id must be included.");
+            }
+        }
+
+#if DEBUG
+        private AnnouncementPatch FakeBody (Query query)
+        {
+            if (!Debugger.IsAttached)
+            {
+                // shouldn't get here.
+                throw new Exception("Phishing attempt.");
+            }
+
+            AnnouncementPatch ap = new AnnouncementPatch
+            {
+                Message = "",
+                Enabled = false,
+                Interval = 1800
+            };
+
+            if (query.ContainsKey("message"))
+            {
+                ap.Message = query["message"];
+            }
+
+            if (query.ContainsKey("enabled"))
+            {
+                bool e = false;
+                if (bool.TryParse(query["enabled"], out e))
+                {
+                    ap.Enabled = e;
+                }
+            }
+
+            if (query.ContainsKey("interval"))
+            {
+                double interval2 = 0d;
+                if (double.TryParse(query["interval"], out interval2))
+                {
+                    ap.Interval = interval2;
```

Comment:
I have no idea what you are doing here or in general with this route, so if it works it works :3